graphbolt/src/cuda/index_select_impl.cu

```diff
@@ -48,6 +53,294 @@ std::pair<torch::Tensor, torch::Tensor> Sort(
   return std::make_pair(sorted_array, sorted_idx);
 }
 
+// Given the in_degree array and a permutation, returns in_degree of the output
+// and the permuted and modified in_degree of the input. The modified in_degree
+// is modified so that there is slack to be able to align as needed.
+template <typename indptr_t, typename indices_t>
+struct AlignmentFunc {
+  static_assert(GPU_CACHE_LINE_SIZE % sizeof(indices_t) == 0);
+  const indptr_t* in_degree;
+  const int64_t* perm;
+  int64_t num_rows;
+  __host__ __device__ auto operator()(int64_t row) {
+    constexpr int num_elements = GPU_CACHE_LINE_SIZE / sizeof(indices_t);
+    return thrust::make_tuple(
+        in_degree[row],
+        (indptr_t)(in_degree[perm ? perm[row % num_rows] : row] + num_elements - 1));
```

Comment:
Could you explain why the indegree should add by `num_elements - 1`?

Reply:
I explained in an inline comment, can you check?